common/oatbox/cache/CacheItem.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/** @noinspection MissingReturnTypeInspection */
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\oatbox\cache;
+
+use DateInterval;
+use DateTime;
+use DateTimeInterface;
+use InvalidArgumentException;
+use Psr\Cache\CacheItemInterface;
+
+class CacheItem implements CacheItemInterface
+{
+    /** @var string */
+    private $key;
+
+    /** @var mixed */
+    private $value;
+
+    /** @var DateTime */
+    private $expiry;
+
+    /** @var bool */
+    private $isHit;
+
+    public function __construct(string $key, bool $hit = false)
+    {
+        $this->key = $key;
+        $this->isHit = $hit;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function get()
+    {
+        return $this->value;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isHit()
+    {
+        return $this->isHit;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function set($value): self
+    {
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function expiresAt($expiration): self
+    {
+        if (null === $expiration) {
+            $this->expiry = null;
+
+            return $this;
+        }
+
+        if ($expiration instanceof DateTimeInterface) {
+            $this->expiry = $expiration->getTimestamp();
+
+            return $this;
+        }
+
+        throw new InvalidArgumentException('Expiration date must implement DateTimeInterface or be null');
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @param DateInterval|null $time
+     */
+    public function expiresAfter($time): self
+    {
+        if (null === $time) {
+            $this->expiry = null;
+
+            return $this;
+        }
+
+        if ($time instanceof DateInterval) {
+            $now = new DateTime('now');
+            $this->expiry = $now->add($time)->getTimestamp();
+
+            return $this;
+        }
+
+        throw new InvalidArgumentException(
```

Comment:
that means in case of int we will throw an `InvalidArgumentException` exception

Reply:
Yes. I did not implement int handling.

Reply:
This is a simple implementation. If we would like to accept integer we need to provide additional validation of this input. This way we cover the most common way to use this. If anybody wants to use an integer instead of DateTime they may just extend this method to provide additional business logic.